common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java

```diff
@@ -101,8 +101,11 @@ public void disconnect() {
      * @return Intent
      */
     public Intent getIntentForAuthService(final Context context) {
-        final String currentActiveBrokerPackageName = getCurrentActiveBrokerPackageName(context);
-        if (currentActiveBrokerPackageName == null || currentActiveBrokerPackageName.length() == 0) {
+        final String currentActiveBrokerPackageName = new BrokerValidator(context).
+                getCurrentActiveBrokerPackageName(context);
+
+        if (currentActiveBrokerPackageName == null || currentActiveBrokerPackageName.length() == 0 ||
+                !isMicrosoftAuthServiceSupported(context.getPackageManager(), currentActiveBrokerPackageName)) {
```

Comment:
Looks like we weren't doing this check before. Why was it NOT needed before? Or why is it needed now?

Reply:
We always did the check, see [here](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java#L137)

Just refactored here to decouple `active broker` check and this